python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -313,20 +340,23 @@ def as_dict(self):
         result.extend(self._extra_metrics.items())
         return dict(result)
 
-    @classmethod
-    def get_metric_keys(cls):
-        """Return a list of metric keys."""
-        return [f.name for f in fields(cls)] + ["cpu_usage", "gpu_usage"]
-
-    @property
+    @metricproperty(
```

Comment:
`metricproperty` is used for two purposes: (1) the regular `@property`  behavior, and (2) initializing the Metric and adding it to the global `_METRICS` variable. what do you think about separating the two, so we can compose the two tags, new `@metric` tag and old `@property` tag? something like:

```
@metric(
    description=...,
    ...
)
@property
def average_num_outputs_per_task(self) -> Optional[float]:
```

Reply:
I originally went with this approach, but decided to combine them. My reasoning was that `@metric` only makes sense to use with properties, and by combining them we use fewer lines of code.

Reply:
If we could use `metric` with non-property methods, I think it makes sense to have separate `metric` and `property` decorators

Reply:
Also I like the consistency/explicitness of `metric_field` + `metric_property` over `metric_field` + `metric`

Reply:
Sounds good for consistency. i don't have a strong preference, the main advantage i had in mind with keeping the two separate is that it's slightly easier to understand the code and gets rid of wrapping logic